pkg/template/template.go

```diff
@@ -163,7 +163,7 @@ func (p *Processor) GenerateParameterValues(t *api.Template) *field.Error {
 		if param.Generate != "" {
 			generator, ok := p.Generators[param.Generate]
 			if !ok {
-				return field.NotFound(templatePath, param)
+				return field.Invalid(templatePath.Child("Generate"), param.Generate, "")
```

Comment:
Invalid takes an additional argument (details about what is invalid about the value)

you can actually copy the message from line 169/170 pretty much.